packages/instructions/src/instruction.ts

```diff
@@ -21,6 +22,28 @@ export interface IInstructionWithAccounts<TAccounts extends readonly (IAccountMe
     readonly accounts: TAccounts;
 }
 
+export function isInstructionWithKnownProgramAddress<TProgramAddress extends string, TInstruction extends IInstruction>(
+    instruction: TInstruction,
+    programAddress: Address<TProgramAddress>,
+): instruction is TInstruction & IInstruction<Address<TProgramAddress>> {
```

Comment:
I would narrow the intersected type here to:

```suggestion
): instruction is TInstruction & { programId: Address<TProgramAddress> } {
```